Remove scalar_check specification and codegen.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30878 Remove TensorImpl::maybe_zero_dim.
* **#30874 Remove scalar_check specification and codegen.**

These have all been disabled at this point, so there is no difference in the generated code.

Differential Revision: [D18855990](https://our.internmc.facebook.com/intern/diff/D18855990)

gchanan committed:
Remove scalar_check specification and codegen.

These have all been disabled at this point, so there is no difference in the generated code.

[ghstack-poisoned]

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

[ghstack-poisoned]

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

[ghstack-poisoned]

albanD commented on c10/core/TensorImpl.h:
```diff
@@ -477,21 +477,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
-   * will become less necessary.  At the moment, it is often called from functions
-   * that correctly handle the 0-dim case, and is just dead code in this case.
-   * In the glorious future, this function will be eliminated entirely.
+   *  DONT CALL THIS.  It will go away very soon.
```
Why does this one remains? It was removed in the opaque version already.

gchanan replied:
Hmm...it's in the commit message but ghstack posted a stale message I think.

Basically, I'm guessing that XLATensor also overrides this so I wanted to be able to get that in without breaking them.

albanD commented on aten/src/ATen/test/broadcast_test.cpp:
```diff
@@ -22,8 +22,7 @@ void TestOut2Basic(DeprecatedTypeProperties& T) {
 void TestOut2WithScalar(DeprecatedTypeProperties& T) {
-  auto aScalar = ones({1}, T);
-  aScalar.unsafeGetTensorImpl()->maybe_zero_dim(true);
+  auto aScalar = ones({}, T);
```
What was the reason to do tests like this? why not create 0-dim tensors directly?

gchanan replied:
That's a good question :).  I can't remember if it was because I was trying to test `maybe_zero_dim` directly or these tests were added before zero-dimensional tensors were exposed publicly so this was the only API.

albanD replied:
Ok !

albanD reviewed (approved):
LGTM

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

Differential Revision: [D18855990](https://our.internmc.facebook.com/intern/diff/D18855990)

[ghstack-poisoned]

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

Differential Revision: [D18855990](https://our.internmc.facebook.com/intern/diff/D18855990)

[ghstack-poisoned]

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

Differential Revision: [D18855990](https://our.internmc.facebook.com/intern/diff/D18855990)

[ghstack-poisoned]

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

Differential Revision: [D18855990](https://our.internmc.facebook.com/intern/diff/D18855990)

[ghstack-poisoned]

gchanan committed:
Update on "Remove scalar_check specification and codegen."


These have all been disabled at this point, so there is no difference in the generated code.

Differential Revision: [D18855990](https://our.internmc.facebook.com/intern/diff/D18855990)

[ghstack-poisoned]
